packages/engine.test/src/testProducer.ts

```diff
@@ -0,0 +1,104 @@
+expect.extend({
+    toMatchExpectations(mockCalls, verbExpectations) {
+        // const remainingExpectations = verbExpectations.slice(0)
+        // const remainingMockCalls = mockCalls.slice(0)
+    
+        const isLonger = mockCalls.length !== verbExpectations.length
+
+        if(isLonger) return {
+            message: () => (`expected calls ${this.utils.printReceived(mockCalls)} to be same length as expectation ${this.utils.printExpected(verbExpectations)}`),
+            pass: false
+        }
+
+        let isSame: boolean = true
+        for(let i = 0; i < mockCalls.length; i++) {
+            expect(mockCalls[i][0]).toEqual(verbExpectations[i])
+
+            // if(verbExpectations[i].match) {
+            //     expect(mockCalls[i][0]).toMatchObject(verbExpectations[i].match)
+            // }
+            // if(verbExpectations[i].equal) {
+            //     expect(mockCalls[i][0]).toEqual(verbExpectations[i].equal)
+            // }
+        }
+
+        // const pass = true
+      
+  
+        // if (pass) {
+        //     return {
+        //         message: () => (`expected ${this.utils.printReceived(received)} not to contain object ${this.utils.printExpected(argument)}`),
+        //         pass: true
+        //     }
+        // } else {
+        //     return {
+        //         message: () => (`expected ${this.utils.printReceived(received)} to contain object ${this.utils.printExpected(argument)}`),
+        //         pass: false
+        //     }
+        // }
+    }
+  })
+
+jest.useFakeTimers();
+
+export function testProducer(name: string, {
+    producer, 
+    presets = {}, 
+    expectations = {}
+}) {
+    test(name, () => {
+        const args = producer.args.value
+        // console.log(args)
+        const keys = Object.keys(args)
+        const mockMap = {}
+        keys.forEach(key => {         
+            switch(args[key].type) {
+                case "OBSERVE":
+                    if(presets[key]) {
+                        mockMap[key] = presets[key]
+                    }
+                    break
+                case "VALUE":
+                    if(presets[key]) {
```

Comment:
How do you handle the following situation?

```
const foo: producer = ({
  a = 123,
  b = Arg.a
})
```
This means `b` will take `a`'s value. But in the current code, `b` will be `undefined`. Unless specified by the developer. I think this case should be factored in as it is possible.

Reply:
this is a feature request, the functionality would be added later on, would have discovered this when trying it on KLP so thanks for pointing that out soon